index.js

```diff
@@ -2,33 +2,37 @@
 const url = require('url')
 const jwt = require('jsonwebtoken')
 
-module.exports = exports = (secret, whitelist) => (fn) => {
-       
+module.exports = exports = (secret, whitelist, config = {}) => (fn) => {
+
     if (!secret) {
         throw Error('micro-jwt-auth must be initialized passing a secret to decode incoming JWT token')
     }
-    
+
+    if (!Array.isArray(whitelist)) {
+        config = whitelist || {}
+    }
+
```

Comment:
I thought it's meaningful to turn `whitelist` to `config` so users don't have to pass an empty array if they want to pass a config but no whitelist.